compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/JvmObservabilityGen.java

```diff
@@ -882,10 +889,17 @@ private void swapBasicBlockContent(BIRBasicBlock firstBB, BIRBasicBlock secondBB
      * @param firstBB The first BB of which terminator should end up in second BB
      * @param secondBB The second BB of which terminator should end up in first BB
      */
-    private void swapBasicBlockTerminator(BIRBasicBlock firstBB, BIRBasicBlock secondBB) {
+    private void swapBasicBlockTerminator(BIRFunction func, BIRBasicBlock firstBB, BIRBasicBlock secondBB) {
         BIRTerminator firstBBTerminator = firstBB.terminator;
         firstBB.terminator = secondBB.terminator;
         secondBB.terminator = firstBBTerminator;
+        int firstBBIndex = func.basicBlocks.indexOf(firstBB);
+        if (firstBBTerminator.lhsOp != null) {
+            BIRBasicBlock endBB = firstBBTerminator.lhsOp.variableDcl.endBB;
```

Comment:
Shall we move this logic to a common method nd reuse? May be passing the `lhsOp` as parameter.